devel/fortio/logger.go

```diff
@@ -0,0 +1,189 @@
+// (this can't be the same name as the pointer receiver version)
+func (l LogLevel) ToString() string {
+	return levelToStrA[l]
+}
+
+// Set is called by the flags.
+func (l *LogLevel) Set(str string) error {
+	var lvl LogLevel
+	var ok bool
+	if lvl, ok = levelToStrM[str]; !ok {
+		// flag processing already logs the value
+		return fmt.Errorf("should be one of %v", levelToStrA)
+	}
+	SetLogLevel(lvl)
+	return nil
+}
+
+// SetLogLevel sets the log level and returns the previous one.
+func SetLogLevel(lvl LogLevel) LogLevel {
+	prev := level
+	if lvl < D {
+		log.Printf("SetLogLevel called with level %d lower than Debug!", lvl)
+		return -1
+	}
+	if lvl > C {
+		log.Printf("SetLogLevel called with level %d higher than Critical!", lvl)
+		return -1
+	}
+	logPrintf(I, "Log level is now %d %s (was %d %s)\n", lvl, lvl.ToString(), prev, prev.ToString())
+	level = lvl
+	return prev
+}
+
+// GetLogLevel returns the current level
+func GetLogLevel() LogLevel {
+	return level
+}
+
+// LogOn returns true if a given level is currently logged.
+func LogOn(lvl LogLevel) bool {
+	return lvl >= level
+}
+
+// LogLevelByName returns the LogLevel by its name.
+func LogLevelByName(str string) LogLevel {
+	return levelToStrM[str]
+}
+
+// Log at the given level.
+// 2 level of calls so it's always same depth for extracting caller file/line
+func Log(lvl LogLevel, format string, rest ...interface{}) {
+	logPrintf(lvl, format, rest...)
+}
+
+func logPrintf(lvl LogLevel, format string, rest ...interface{}) {
+	if !LogOn(lvl) {
+		return
+	}
+	if *logFileAndLine {
+		_, file, line, _ := runtime.Caller(2)
+		file = file[strings.LastIndex(file, "/")+1:]
+		log.Print(levelToStrA[lvl][0:1], " ", file, ":", line, *logPrefix, fmt.Sprintf(format, rest...))
+	} else {
+		log.Print(levelToStrA[lvl][0:1], " ", *logPrefix, fmt.Sprintf(format, rest...))
+	}
+	if lvl == F {
+		panic("aborting...")
+	}
+}
+
+// -- would be nice to be able to create those in a loop instead of copypasta:
+
+// Dbg logs if Debug level is on.
+func Dbg(format string, rest ...interface{}) {
+	logPrintf(D, format, rest...)
+}
+
+// LogV logs if Verbose level is on.
+func LogV(format string, rest ...interface{}) {
+	logPrintf(V, format, rest...)
+}
+
+// Info logs if Info level is on.
+func Info(format string, rest ...interface{}) {
+	logPrintf(I, format, rest...)
+}
+
+// Warn logs if Warning level is on.
+func Warn(format string, rest ...interface{}) {
+	logPrintf(W, format, rest...)
+}
+
+// Err logs if Warning level is on.
+func Err(format string, rest ...interface{}) {
+	logPrintf(E, format, rest...)
+}
+
+// Crit logs if Warning level is on.
+func Crit(format string, rest ...interface{}) {
+	logPrintf(C, format, rest...)
+}
+
+// Fatal logs if Warning level is on.
+func Fatal(format string, rest ...interface{}) {
+	logPrintf(F, format, rest...)
+}
+
+// DbgOn is a shortcut for LogOn(D).
+func DbgOn() bool {
+	return LogOn(D)
+}
+
+// VOn is a shortcut for LogOn(V).
+func VOn() bool {
```

Comment:
it's `DbgOn()` because the logging shortcut is `Dbg()`
`V()` was a bit short for a function and `Verbose()` a bit long and `Verb()` or `Vrb()` not clear or misleading so I called the shortcut for V level logging `LogV()`
so I guess this should in theory be LogVOn() or IsLoggingV()... not sure... naming is hard :-)